config/samples/olm_v1alpha1_clusterextension.yaml

```diff
@@ -28,7 +28,11 @@ kind: ClusterRole
 metadata:
   name: argocd-installer-clusterrole
 rules:
-# Manage ArgoCD CRDs 
+# Allow ClusterExtension to set blockOwnerDeletion ownerReferences
+- apiGroups: [olm.operatorframework.io]
+  resources: [clusterextensions/finalizers]
+  verbs: [update]
```

Comment:
Technically, this is the only necessary change to RBAC.

The other RBAC changes are cleanups that I missed in #1102

Reply:
Is this a permission that should need to be set by a user of a ClusterExtension or is this a permission we should ensure is available on all ClusterExtensions by operator-controller's ServiceAccount?

Reply:
It just seems a bit of an odd UX to me to have a user need to provide permissions for the ClusterExtension controller to do things with a ClusterExtension

Reply:
The plugin says: "The user creating an object that contains an ownerReference with `blockOwnerDeletion: true`, must have permission to control the owner's deletion configuration (i.e. to set finalizers on the owner)"

The user in this case has to be the ClusterExtension service account because it is the one creating these objects that have these owner refs back to the ClusterExtension.

Reply:
Ah I see. That makes sense. I'm not familiar with the plugin and found a brief reference to it. Could you update the PR description to include this information? Is there a way we could document this behavior as well so that users know that when this plugin is enabled they _have_ to add this to the ServiceAccount being referenced in the ClusterExtension?

I'm fine with the documentation being a follow-up, just would like to make sure we have a tracker for it somewhere to ensure we don't forget.

Reply:
Added a draft doc for this in this PR!